### PR TITLE
Painting fence || DP ||documentation

### LINKED ISSUE #154: documentation for painting fence problem
## 🚀 Feature

Given a fence with n posts and k colors, find out the number of ways of painting the fence such that at most 2 adjacent posts have the same color. 

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

Documentation for painting fence problem

## Pitch

Documentation for painting fence problem

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #154 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #154

### CODE REVIEW

File: docs/dynamic-programming/painting-fence.mdx [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+title: Painting Fence
+slug: /mathematics/painting-fence
+---
+
+## Introduction
+
+Given a fence with `n` posts and `k` colors , find out the number of ways of painting the fence such that atmost `2` adjacent posts have the same color.
```
Comment (HarshCasper) [inline docs/dynamic-programming/painting-fence.mdx:8]: Would appreciate a bit more explanation here

File: docs/dynamic-programming/painting-fence.mdx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+## Explanation
+
+Suppose we have a single post, The number of ways of painting the fence with `k` colors is `k` Now we consider `2` posts, Then the number of ways of painting the second fence with the same color as the first post is `k`, and the number of ways of painting the second post with a different color is `k*(k-1)` as we have `k-1` colors that are different from the first post's color. So the total number of ways of painting `2` fences is `k*(k-1) + k`. Now we consider `3` posts, Then the number of ways of painting the third fence same as the second fence is the number of ways of painting the second fence with a different color that is `k*(k-1)`. The number of ways of painting the third fence using a different color is equal to the `total number of ways of painting the second fence * (k-1)`. Hence the total number of ways of painting `3` fences is `k*(k-1) + (k+k*(k-1))*(k-1)`. This way we can find the total number of ways of painting n fences
```
Comment (HarshCasper) [inline docs/dynamic-programming/painting-fence.mdx:12]: Better break this down into consumable bits. Also don't use back-tick formatting indiscriminately

File: docs/dynamic-programming/painting-fence.mdx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    - not possible 
+- declare three arrays `total`,`same`,`diff` of length `n-1` 
+- initialise `same[0]` to `k`
+- initialise `diff[0]` to `k*(k-1)`
+- initialise `total[0]` to `k*k`
+- loop from `i = 1` to `n-1`
+    - `same[i] = diff[i-1]`
+    - `diff[i] = total[i-1]*(k-1)`
+    - `total[i] = same[i] + diff[i]`
+- return `total[n-2]`
```
Comment (HarshCasper) [inline docs/dynamic-programming/painting-fence.mdx:26]: Don't use formulas here. Better use Text

File: docs/dynamic-programming/painting-fence.mdx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+#### Output
+
+```
+Number of Possiblities: 120
+```
+
+## Complexity Analysis
+
+- Time complexity - O(`n`)
+- Space complexity - O(`n`), where `n` is the number of posts
```
Comment (HarshCasper) [inline docs/dynamic-programming/painting-fence.mdx:136]: Use backticks around the entire value

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  Can you please review and merge my pr, I have made the changes

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**